### PR TITLE
Add more colour options

### PR DESCRIPTION
**Pull request by mariabnd**
Add green to main palette and colours from figs 3.4-3.5, 3.17, 3.26, and 3.29-3.31

### CODE REVIEW

**Comment by seabbs** [comment]
This is great @mariabnd and happy to merge.

Could you add some information to the palettes section of the docs on what palettes are now supported + what they are and how many colours are in each?

Also feel free to add an entry to the news section (giving yourself credit) :smile:

**Comment by mariabnd** [comment]
Yep, will try to have that sorted by end of tomorrow

**Comment by seabbs** [comment]
Making any headway with this @mariabnd? Happy to look at if you would like.

**Comment by mariabnd** [comment]
Sorry about that. I was wondering whether it would make more sense to include an `n` argument allowing the user to select the number of colours they would like included in a palette (similar to what [ECDC is doing](https://github.com/EU-ECDC/EcdcColors)), rather than have `greens`, `greens2`, etc. Do you have any thoughts on this?

**Comment by seabbs** [comment]
No problem.

Good idea! Sounds like a much cleaner approach.

**Comment by mariabnd** [comment]
Do you agree with ECDC's view that no more than seven colours should be used?

**Comment by seabbs** [comment]
I think that makes sense to start with - given that presumably most people using the palette won't need more than 7 discrete colours. 

It could be something to revisit down the line though. 

Also happy to give users the option (there isn't any reason not too) of more colours if you are happy to implement.

**Comment by mariabnd** [comment]
I've made a simple fix (adding an `n` argument which defaults to `NULL`; all colours). If we want to ensure spacing of colours, it might make sense to import palettes from [EcdcColors](https://github.com/EU-ECDC/EcdcColors) (for greens, blues, and reds at least).

**Comment by seabbs** [comment]
Thanks for this. 

I am very with this solution but also happy for it to be extended to be more complete.

Not very keen on importing colours from {EcdcColors} as want to avoid adding an additional dependency (and can't add one that isn't on CRAN). 

I'm seeing some check problems from this pull request (below). 

```
> plot_tb_burden_summary(countries = "United Kingdom", compare_all_regions = FALSE,
                                               compare_to_region = TRUE, conf = NULL, verbose = FALSE) +
   theme_who() +
   scale_colour_who(reverse = TRUE)
Error in if (n > length(pal)) stop("n is larger than number of available colours 
for chosen palette") argument is of length zero
```

This looks like an issue with the order of your error checking. My recommendation would be to do: 

```
if (is.null(n)) {
n <- length(pal)
}

 if (n > length(pal)) {
     stop("n is larger than number of available colours for chosen palette")
}else{
 pal <- pal[1:n]
}
```

Potentially could add some better colour selection than `pal <- pal[1:n]`. 


Final things before merging are: 

1. Could you add a test for this to check the functionality of the n argument. 
2. `scale_fill_who` and `scale_colour_who` both need an n argument that is then passed to the WHO palette.

**Comment by mariabnd** [comment]
Alright! So by the end of this week I will
- [ ] Incorporate recommendation from above
- [ ] Extract all colours used in the TB report
- [ ] Provide n argument to scale functions
- [ ] Add tests

**Comment by seabbs** [comment]
Sounds good!

**Comment by mariabnd** [comment]
Have updated such that the theme now has ten `palette` options. Tests to follow

**Comment by seabbs** [comment]
Just been checking out the travis issues.

 It looks like you need to update the documentation (`devtools::document()`) so that the code/docs match.

File: tests/testthat/test-who_palettes.R [hunk 1]
```diff
@@ -0,0 +1,10 @@
+context("who_palettes")
+
+test_that("who_palettes cannot take n larger than palettes or theme not provided", {
+  
+  large_n <- who_palettes(palette = "main", n = 100)
+  expect_error(large_n, "n is larger than number of available colours for chosen palette")4
```
Comment (seabbs) [inline tests/testthat/test-who_palettes.R]: Seeing type at the end of the line (4) causing this to fail.

File: tests/testthat/test-who_palettes.R [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  
+  wrong_col <- who_palettes(palette = "dark")
+  expect_error(wrong_col, "chosen palette not available")
```
Comment (seabbs) [inline tests/testthat/test-who_palettes.R]: I think this needs to be in a second test.

**Comment by seabbs** [comment]
No rush but would be cool to get this sorted before the new year. 

Happy to help out if you would like?

**Comment by mariabnd** [comment]
Yes please! Things got really busy all of a sudden

**Comment by seabbs** [comment]
Sorry I didn't have a chance to get to this over the Christmas break. It would be great to get this wrapped up by the end of the month as I am planning on knocking off a few more issues and going for a CRAN release. 

Are you interested (do you have the time!) to wrap this up or shall I look at doing so?

File: tests/testthat/test-who_palettes.R (hunk 1 above)
Comment (mariabnd) [reply tests/testthat/test-who_palettes.R]: I removed the 4 but that doesn't seem to have solved the problem

**Comment by seabbs** [comment]
So looking at Travis I am seeing the following error: 

```
  ══ testthat results  ═══════════════════════════════════════════════════════════
1293  [ OK: 146 | SKIPPED: 21 | WARNINGS: 9 | FAILED: 1 ]
1294  1. Error: who_palettes cannot take n larger than palettes or theme not provided (@test-who_palettes.R#5) 
1295  
```

I think all you need to do is move the function call into expect error. Ideally, the functions themselves should then print the informative error messages rather than relying on testthat (in case this happens to the user). 


Something like this: 


```
context("who_palettes")

test_that("who_palettes cannot take n larger than palettes", {
  expect_error(who_palettes(palette = "main", n = 100))
  }) 


test_that("who_palettes fails if a unrecognised palette is supplied", {
  expect_error(who_palettes(palette = "dark"))
})
```

Sorry I missed that before.

**Comment by mariabnd** [comment]
Brilliant! Thanks for finding the solution

**Comment by seabbs** [comment]
No problem. I will just let the checks run, review everything again, and then merge. Thanks for adding this!

**Pull request merged**